Cargo.toml

```diff
@@ -21,6 +21,10 @@ rand = "0.6.5"
 byteorder = "1.3.1"
 failure = "0.1.5"
 subtle = "2.0.0"
+sha-1 = "0.8.1"
```

Comment:
I would really like to avoid these dependencies, especially as I want to change signing to be abstract over `Digest`. See  https://github.com/RustCrypto/signatures/pull/7 for some details on the plans there

Reply:
Ok I agree that depending on every hashing crates is not a good idea. The digest implementation should be passed. I tried that in the first place but failed, fighting the Rust compiler due to my limited Rust experience, I don't remember the exact reason...
I'll see what I can do (see my other comment on my commitment)